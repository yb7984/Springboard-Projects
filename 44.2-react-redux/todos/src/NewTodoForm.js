import TodoForm from './TodoForm';

const NewTodoForm = ({ addTodo }) => {

    return (<TodoForm id="" todo="" updateTodo={addTodo} />);
}

export default NewTodoForm;