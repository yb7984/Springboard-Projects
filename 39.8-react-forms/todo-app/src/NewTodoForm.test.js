import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("renders without crashing", function () {
  render(<NewTodoForm />);
});


it("matches snapshot, show empty input", function () {
  const { asFragment , queryByLabelText} = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();

  const inputTodo = queryByLabelText("New Todo:");

  expect(inputTodo).toBeInTheDocument();
  expect(inputTodo.value).toBe("");
});