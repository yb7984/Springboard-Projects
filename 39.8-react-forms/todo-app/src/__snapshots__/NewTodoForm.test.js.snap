// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, show empty input 1`] = `
<DocumentFragment>
  <div
    class="TodoForm"
  >
    <form>
      <label
        for="todo-"
      >
        New Todo:
      </label>
      <input
        id="todo-"
        name="todo"
        placeholder="Things to do"
        required=""
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
